router/gin/endpoint_test.go

```diff
@@ -110,7 +125,13 @@ func setup(timeout time.Duration, p proxy.Proxy) (string, *http.Response, error)
 func startGinServer(handlerFunc gin.HandlerFunc) *gin.Engine {
 	gin.SetMode(gin.TestMode)
 	router := gin.New()
-	router.GET("/_gin_endpoint/:param", handlerFunc)
+	router.GET("/_gin_endpoint/:param", ctxMiddleware, handlerFunc)
```

Comment:
Not related to the PR itself, but:

> At Google, we require that Go programmers pass **a Context parameter as the first argument** to every function on the call path between incoming and outgoing requests. This allows Go code developed by many different teams to interoperate well. It provides simple control over timeouts and cancelation and ensures that critical values like security credentials transit Go programs properly.

Do you think it makes sense to open a PR and to set this standard in Krakend as well?

Reply:
this is not a context. it's a dummy gin middleware just for adding some values in the gin context and check if they are present at the proxy layer.

but if you find any call not following the standard, please create as many PR as required to fix all of them 😉